src/librustdoc/clean/types.rs

```diff
@@ -49,18 +48,105 @@ use self::ItemKind::*;
 use self::SelfTy::*;
 use self::Type::*;
 
-thread_local!(crate static MAX_DEF_IDX: RefCell<FxHashMap<CrateNum, DefIndex>> = Default::default());
+crate type DefIdSet = FxHashSet<DefId>;
+crate type RealDefId = rustc_hir::def_id::DefId;
```

Comment:
Instead of making an alias, shouldn't we do the opposite and instead rename the `DefId` enum into `DefIdWrapper` or something along the line?

Reply:
What would be the advantage of a wrapper type? `RealDefId` is just a rustc `DefId` without any extras

Reply:
And I suggested the opposite: using `DefId` for "normal" cases and rename the enum to make it obvious that it's not a `DefId` but might be fake.

Reply:
So instead of having `RealDefId` and `DefId`, you would have `DefId` (the real one) and `FakeDefId` (which might be fake)?

I don't think that really matters and is only a naming preference, because you would just use `DefId` instead of `RealDefId` and `FakeDefId` instead of `DefId`, but I'm also fine with changing it to your suggestion.

Reply:
Well, at least it's more coherent because currently, `DefId` isn't `DefId`. I think it's confusing and not really helpful when reading such huge code base.

Reply:
And thanks if you update it! I really think it'd improve the readability. :)

Reply:
I agree with Guillaume here: I would like rustdoc code to be as similar to compiler (and compiler docs!) code as possible to avoid confusion. We should have DefId stay DefId and introduce FakeDefId.